### DISCUSSION TITLE
Isn't SPA experience very poor when using getServerSideProps?

### DISCUSSION THREAD

**Discussion by borjanavarro** [vercel/next.js#26201] [Help]
I'm wondering how can achieve the dynamic loadings and change of state the same way like react hooks. Maybe I'm using wrong the framework but I feel that I have completely lose the "dynamicness".  The need of having all data in server-side so as search engines can take it breaks all we had achieved with react. Can anybody tell how to get the best of the two worlds?

**Comment by leerob** [comment]
https://github.com/vercel/next.js/discussions/29117#discussioncomment-6870762
